### DISCUSSION TITLE
Manual Validation is not working for elements looped in an array (@bind-Text=address[iteration])

### DISCUSSION THREAD

**Discussion by DurgaSanthi** [Megabit/Blazorise#2559] [General]
Blazorise - Manual Validation is not working because of array value used to bind the text. Example: @bind-Text="address[iteration]"

This works for auto validation but not for manual validation

<Validations @ref="validations" Mode="ValidationMode.Manual">

Created TextEdit components in loop. For example there are 10 elements and Element Id = addressid[0], addressid[1] etc.
And the @bind-Text="address[0]" etc

Below is the exception am getting.

Exception message:

Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
Unhandled exception rendering component: The provided expression contains a InstanceMethodCallExpression1 which is not supported. FieldIdentifier only supports simple member accessors (fields, properties) of an object.
System.ArgumentException: The provided expression contains a InstanceMethodCallExpression1 which is not supported. FieldIdentifier only supports simple member accessors (fields, properties) of an object.
at Microsoft.AspNetCore.Components.Forms.FieldIdentifier.ParseAccessor[String](Expression1 accessor, Object& model, String& fieldName) at Microsoft.AspNetCore.Components.Forms.FieldIdentifier.Create[String](Expression1 accessor)
at Blazorise.Validation.InitializeInputExpression[String](Expression`1 expression)
at Blazorise.TextEdit.SetParametersAsync(ParameterView parameters)

Can someone help

**Comment by stsrki** [comment]
I would say you use the `Validator` parameter for validations? Can you share you code so we can reproduce it on our side?

**Reply by DurgaSanthi** [reply]
****

```cshtml
<Validations @ref="validations" Mode="ValidationMode.Manual">
<Validation UsePattern="true">
	<Field Class="col-3">
		<FieldBody>
			<TextEdit ElementId="@addressId[0]" Class="form-control form-control-xs" **@bind-Text="address[iteration]"** ChangeTextOnKeyPress="false" Placeholder="Address" @ref="@addressRef[iteration]" Pattern="^[A-Za-z0-9]{12}$" MaxLength="12"></TextEdit>
		</FieldBody>
	</Field>
</Validation>
<Validations>

```

It throws an exceptions. When I tested I realized that is because of @bind-Text="address[iteration]".

Please suggest.

Thank you.

**Reply by stsrki** [reply]
Do you have `for` loop?

**Reply by DurgaSanthi** [reply]
```cshtml
for (int iteration = 0; iteration < maxNum; iteration++)
{
	<Validations @ref="validations" Mode="ValidationMode.Manual">
	<Validation UsePattern="true">
		<Field Class="col-3">
			<FieldBody>
				<TextEdit ElementId="@addressId[iteration]" Class="form-control form-control-xs" @bind-Text="address[iteration]" ChangeTextOnKeyPress="false" Placeholder="Address" @ref="@addressRef[iteration]" Pattern="^[A-Za-z0-9]{12}$" MaxLength="12"></TextEdit>
			</FieldBody>
		</Field>
	</Validation>
	<Validations>
}
```

**Reply by stsrki** [reply]
There were several issues

- `Validations` was repeated for every item, it must be only one
- your `maxNum` and `address` list were _probably_ not the same size
- most important, an array item cannot be used for two-way binding. That's why it gave you an error

I optimized your code and created a class for holding all information like element id and reference, Also you can see the `@key` attribute. It is very important for the Blazor diffing algorithm.

And you don't need to manually write `"form-control form-control-xs"` or `"col-3"`. Blazorise will handle those for you.

```cshtml
<Validations @ref="validations" Mode="ValidationMode.Manual">
    <Fields>
        @foreach ( var address in addressList ?? Enumerable.Empty<AddressInfo>() )
        {
            <Validation UsePattern="true">
                <Field ColumnSize="ColumnSize.Is3">
                    <FieldBody>
                        <TextEdit @key="@address" @ref="@address.TextEditRef" ElementId="@address.ElementId" @bind-Text="address.Address" Size="Size.Small" ChangeTextOnKeyPress="false" Placeholder="Address" Pattern="^[A-Za-z0-9]{12}$" MaxLength="12" />
                    </FieldBody>
                </Field>
            </Validation>
        }
    </Fields>
</Validations>
<Button Color="Color.Primary" Clicked="@(()=>validations.ValidateAll())">
    Submit
</Button>
@code{
    class AddressInfo
    {
        public TextEdit TextEditRef { get; set; }

        public string ElementId { get; set; }

        public string Address { get; set; }
    }

    Validations validations;

    AddressInfo[] addressList;

    [Inject] public IIdGenerator IdGenerator { get; set; }

    override protected void OnInitialized()
    {
        addressList = Enumerable.Range( 1, 5 ).Select( x => new AddressInfo
        {
            ElementId = IdGenerator.Generate,
        } ).ToArray();
    }
}
```

**Reply by DurgaSanthi** [reply]
Thanks a lot. Let me try.
